packages/shared/src/utils.ts

```diff
@@ -65,10 +71,19 @@ export function buildEntityPath(
   return `${prefix}/${entityNameOrRelativePath}`;
 }
 
-export function createArrayFromView<T>(view: NdArray<T[]>): NdArray<T[]> {
-  const array = ndarray<T[]>([], view.shape);
-  assign(array, view);
+export function createArrayFromView<T, U extends TypedArray | T[]>(
+  view: NdArray<U>
+): NdArray<U> {
+  const { data } = view;
+
+  const array = ndarray(
+    // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
+    // @ts-ignore (`data.constructor` resolves to `Function`)
+    (isTypedArray(data) ? new data.constructor(view.size) : []) as U,
+    view.shape
+  );
```

Comment:
If the input ndarray is typed, `createArrayFromView` must now initialize an empty ndarray with the same typed array (and buffer length). `new data.constructor()` works perfectly fine, but was too challenging to type because of the generic.